x-pack/plugins/monitoring/server/lib/alerts/fetch_missing_data.ts

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { get } from 'lodash';
+import { AlertCluster, AlertMissingData } from '../../alerts/types';
+import {
+  KIBANA_SYSTEM_ID,
+  BEATS_SYSTEM_ID,
+  APM_SYSTEM_ID,
+  LOGSTASH_SYSTEM_ID,
+  ELASTICSEARCH_SYSTEM_ID,
+} from '../../../common/constants';
+
+interface IndexBucketESResponse {
+  key: string;
+  clusters: {
+    buckets: ClusterBucketESResponse[];
+  };
+}
+
+interface ClusterBucketESResponse {
+  key: string;
+  kibana_uuids: UuidResponse;
+  logstash_uuids: UuidResponse;
+  es_uuids: UuidResponse;
+  beats: {
+    beats_uuids: UuidResponse;
+  };
+  apms: {
+    apm_uuids: UuidResponse;
+  };
+}
+
+interface UuidResponse {
+  buckets: UuidBucketESResponse[];
+}
+
+interface UuidBucketESResponse {
+  key: string;
+  most_recent: {
+    value: number;
+  };
+  document: {
+    hits: {
+      hits: TopHitESResponse[];
+    };
+  };
+}
+
+interface TopHitESResponse {
+  _source: {
+    source_node?: {
+      name?: string;
+    };
+    kibana_stats?: {
+      kibana?: {
+        name?: string;
+      };
+    };
+    logstash_stats?: {
+      logstash?: {
+        host?: string;
+      };
+    };
+    beats_stats?: {
+      beat?: {
+        name?: string;
+      };
+    };
+  };
+}
+
+function findNonEmptyBucket(bucket: ClusterBucketESResponse): UuidResponse {
+  if (bucket.beats.beats_uuids.buckets.length > 0) {
+    return bucket.beats.beats_uuids;
+  }
+  if (bucket.apms.apm_uuids.buckets.length > 0) {
+    return bucket.apms.apm_uuids;
+  }
+  if (bucket.kibana_uuids.buckets.length > 0) {
+    return bucket.kibana_uuids;
+  }
+  if (bucket.logstash_uuids.buckets.length > 0) {
+    return bucket.logstash_uuids;
+  }
+  if (bucket.es_uuids.buckets.length > 0) {
+    return bucket.es_uuids;
+  }
+  return { buckets: [] };
+}
+
+function getStackProductFromIndex(index: string, bucket: ClusterBucketESResponse) {
+  if (index.includes('-kibana-')) {
+    return KIBANA_SYSTEM_ID;
+  }
+  if (index.includes('-beats-')) {
+    if (bucket.apms.apm_uuids.buckets.length > 0) {
+      return APM_SYSTEM_ID;
+    }
+    return BEATS_SYSTEM_ID;
+  }
+  if (index.includes('-logstash-')) {
+    return LOGSTASH_SYSTEM_ID;
+  }
+  if (index.includes('-es-')) {
+    return ELASTICSEARCH_SYSTEM_ID;
+  }
+  return '';
+}
+
+export async function fetchMissingData(
+  callCluster: any,
+  clusters: AlertCluster[],
+  index: string,
+  limit: number,
+  size: number
+): Promise<AlertMissingData[]> {
+  const endMs = +new Date();
+  const startMs = endMs - limit - limit * 0.25; // Go a bit farther back because we need to detect the difference between seeing the monitoring data versus just not looking far enough back
```

Comment:
I don't think this is a good idea, since 25% is a pretty big padding for the one day default. How about we just have a minimum limit of 3 minutes? That way we can do something like: `endMs - (limit + 180000)`

Reply:
Yea, that's probably fair. I guess I just wanted to be sure I accounted for various changes to the default collection period  but `3m` is probably a good enough distance to go back. Thanks!